File: 500.600/521.LongestUncommonSubsequence.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

class LongestUncommonSubsequence
{
public:
    int findLUSlength(string a, string b)
    {
        int m = a.size();
        int n = b.size();
        if (m != n || a != b)
        {
            return max(m, n);
        }
        return -1;
    }
};

int main()
{
    LongestUncommonSubsequence cls;
    cout << cls.findLUSlength("aba", "cdc") << " -> 3" << endl;
    cout << cls.findLUSlength("aaa", "bbb") << " -> 3" << endl;
    cout << cls.findLUSlength("aaa", "aaa") << " -> -1" << endl;
    return 0;
}

File: go/500.600/521.LongestUncommonSubsequence.go
package main

import (
	"fmt"
)

func max(i, j int) int {
	if i > j {
		return i
	}
	return j
}

func findLUSlength(a string, b string) int {
	if a == b {
		return -1
	}
	return max(len(a), len(b))
}

func main() {
	fmt.Println(findLUSlength("aba", "cdc"), 3)
	fmt.Println(findLUSlength("aaa", "bbb"), 3)
	fmt.Println(findLUSlength("aaa", "aaa"), -1)
}
